AR-Top-MobileApp/Assets/Prefabs/MapController.cs

```diff
@@ -234,41 +239,69 @@ void buildPiece (GridModel obj)
 			colorize (tilePiece, obj.color);
+			rayCastText = tilePiece.GetComponent <RaycastText> ();
+			rayCastText.clearRaycastInfoText ();
+			rayCastInfo = "(" + obj.position.x.ToString () + ", " + obj.position.z.ToString () + ", " + (obj.position.y + 1).ToString () + ")";
+			rayCastText.appendRaycastInfoText (rayCastInfo);
 			break;
 		case "floor":
 			tilePiece = Instantiate (floorPrefab, tileVector, Quaternion.identity);
 			tilePiece.transform.SetParent (modelLayer.transform);
 			colorize (tilePiece, obj.color);
+			rayCastText = tilePiece.GetComponent <RaycastText> ();
+			rayCastText.clearRaycastInfoText ();
+			rayCastInfo = "(" + obj.position.x.ToString () + ", " + obj.position.z.ToString () + ", " + (obj.position.y + 1).ToString () + ")";
+			rayCastText.appendRaycastInfoText (rayCastInfo);
 			break;
 		case "fighter":
 			tilePiece = Instantiate (fighterPrefab, tileVector, Quaternion.identity);
 			tilePiece.transform.SetParent (baseLayer.transform);
 			basePiece = tilePiece.transform.GetChild (1);
 			colorize (basePiece, obj.color);
+			rayCastText = tilePiece.GetComponent <RaycastText> ();
+			rayCastText.clearRaycastInfoText ();
+			rayCastInfo = "(" + obj.position.x.ToString () + ", " + obj.position.z.ToString () + ", " + (obj.position.y + 1).ToString () + ")";
+			rayCastText.appendRaycastInfoText (rayCastInfo);
 			break;
 		case "knight":
 			tilePiece = Instantiate (knightPrefab, tileVector, Quaternion.identity);
 			tilePiece.transform.SetParent (baseLayer.transform);
 			basePiece = tilePiece.transform.GetChild (1);
 			colorize (basePiece, obj.color);
+			rayCastText = tilePiece.GetComponent <RaycastText> ();
+			rayCastText.clearRaycastInfoText ();
+			rayCastInfo = "(" + obj.position.x.ToString () + ", " + obj.position.z.ToString () + ", " + (obj.position.y + 1).ToString () + ")";
+			rayCastText.appendRaycastInfoText (rayCastInfo);
 			break;
 		case "ranger":
 			tilePiece = Instantiate (rangerPrefab, tileVector, Quaternion.identity);
 			tilePiece.transform.SetParent (baseLayer.transform);
 			basePiece = tilePiece.transform.GetChild (1);
 			colorize (basePiece, obj.color);
+			rayCastText = tilePiece.GetComponent <RaycastText> ();
+			rayCastText.clearRaycastInfoText ();
+			rayCastInfo = "(" + obj.position.x.ToString () + ", " + obj.position.z.ToString () + ", " + (obj.position.y + 1).ToString () + ")";
+			rayCastText.appendRaycastInfoText (rayCastInfo);
 			break;
 		case "goblin":
 			tilePiece = Instantiate (goblinPrefab, tileVector, Quaternion.identity);
 			tilePiece.transform.SetParent (baseLayer.transform);
 			basePiece = tilePiece.transform.GetChild (1);
 			colorize (basePiece, obj.color);
+			rayCastText = tilePiece.GetComponent <RaycastText> ();
+			rayCastText.clearRaycastInfoText ();
+			rayCastInfo = "(" + obj.position.x.ToString () + ", " + obj.position.z.ToString () + ", " + (obj.position.y + 1).ToString () + ")";
+			rayCastText.appendRaycastInfoText (rayCastInfo);
 			break;
 		}
```

Comment:
Should probably add a default case to this switch block which exits the function early.

Reply:
Yeah I was wondering about that. Do you want the default to just be a voxel?

Reply:
No, we should probably throw an exception if the type is not recognized.